tools/developer/lps2jani/lps2jani.cpp

```diff
@@ -0,0 +1,452 @@
+   */
+  array convert_action_declarations(const process::action_label_list& act_decls) const
+  {
+    array actions;
+    for (const process::action_label& d : act_decls)
+    {
+      actions.push_back(object{ {"name", (std::string)d.name()} });
+    }
+    return actions;
+  }
+
+  /**
+   * \brief Converts the given OpId into the JSON string that corresponds to it according to the JANI schema.
+   *
+   * \param opid the to-be converted OpId.
+   * \return A JSON string corresponding to the operator.
+   */
+  string convert_operator_to_jani(const data::data_expression& opid) const {
+    if (std::string op = mcrl2::data::pp(opid);
+      "!=" == op) {
+      return "≠";
+    }
+    else if ("==" == op) {
+      return "=";
+    }
+    else if ("<=" == op) {
+      return "≤";
+    }
+    else if ("&&" == op) {
+      return "∧";
+    }
+    else if ("||" == op) {
+      return "∨";
+    }
+    else if ("@cReal" == op) { // threat real numbers as a division operator
+      return "/";
+    }
+    else {
+      return op;
+    }
+  }
+
+  /**
+   * Converts the given data expression into a corresponding JSON value in line with the JANI Expression schema.
+   *
+   * \param e_in the to-be converted expression.
+   * \return A JSON value corresponding to the expression.
+   */
+  value convert_data_expression(const data::data_expression& e_in) const
+  {
+    const data::data_expression e = m_data_structures->rewriter(e_in);
+    if (is_variable(e))
+    {
+      return (std::string)atermpp::down_cast<data::variable>(e).name();
+    }
+    else if (data::sort_pos::is_positive_constant(e) ||
+      data::sort_nat::is_natural_constant(e) ||
+      data::sort_int::is_integer_constant(e)) {
+      return std::stoi(pp(e));
+    }
+    else if (data::sort_bool::is_true_function_symbol(e)) {
+      return true;
+    }
+    else if (data::sort_bool::is_false_function_symbol(e)) {
+      return false;
+    }
+    else if (data::sort_bool::is_not_application(e))
+    {
+      const data::application& appl = atermpp::down_cast<data::application>(e);
+      return object{
+        {"op", "¬"},
+        {"exp", convert_data_expression(appl[0])}
+      };
+    }
+    else if (is_greater_application(e)) // > is not supported within jani, and thus should be flipped
+    {
+      const data::application& appl = atermpp::down_cast<data::application>(e);
+      return object{
+        {"left", convert_data_expression(appl[1])},
+        {"op", "<"},
+        {"right", convert_data_expression(appl[0])}
+      };
+    }
+    else if (is_greater_equal_application(e)) // >= is not supported within jani, and thus should be flipped
+    {
+      const data::application& appl = atermpp::down_cast<data::application>(e);
+      return object{
+        {"left", convert_data_expression(appl[1])},
+        {"op", "≤"},
+        {"right", convert_data_expression(appl[0])}
+      };
+    }
+    else if (data::sort_bool::is_implies_application(e)) {
```

Comment:
Parentheses are inconsistently put on the next line vs after the conditional. I don't really mind putting it after for a one line code, but at least it should be consistently applied.